docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,381 @@
+RPM-DEPENDENCY-GENERATORS(7)
+
+# NAME
+*rpm-dependency-generators* - RPM file attribute and dependency generator
+interface
+
+# SYNOPSIS
+
+## Generators
+*%\_\_*_NAME_*\_conflicts*[*()*] _BODY_++
+*%\_\_*_NAME_*\_enhances*[*()*] _BODY_++
+*%\_\_*_NAME_*\_obsoletes*[*()*] _BODY_++
+*%\_\_*_NAME_*\_orderwithrequires*[*()*] _BODY_++
+*%\_\_*_NAME_*\_provides*[*()*] _BODY_++
+*%\_\_*_NAME_*\_recommends*[*()*] _BODY_++
+*%\_\_*_NAME_*\_requires*[*()*] _BODY_++
+*%\_\_*_NAME_*\_suggests*[*()*] _BODY_++
+*%\_\_*_NAME_*\_supplements*[*()*] _BODY_
+
+## Matching rules
+*%\_\_*_NAME_*\_path* _REGEX_++
+*%\_\_*_NAME_*\_magic* _REGEX_++
+*%\_\_*_NAME_*\_exclude_path* _REGEX_++
+*%\_\_*_NAME_*\_exclude_magic* _REGEX_
+
+## Tunables
+*%\_\_*_NAME_*\_conflicts_opts* _BODY_++
+*%\_\_*_NAME_*\_enhances_opts* _BODY_++
+*%\_\_*_NAME_*\_obsoletes_opts* _BODY_++
+*%\_\_*_NAME_*\_orderwithrequires_opts* _BODY_++
+*%\_\_*_NAME_*\_provides_opts* _BODY_++
+*%\_\_*_NAME_*\_recommends_opts* _BODY_++
+*%\_\_*_NAME_*\_requires_opts* _BODY_++
+*%\_\_*_NAME_*\_suggests_opts* _BODY_++
+*%\_\_*_NAME_*\_supplements_opts* _BODY_
+
+*%\_\_requires_exclude* _REGEX_++
+*%\_\_requires_exclude_from* _REGEX_++
+*%\_\_provides_exclude* _REGEX_++
+*%\_\_provides_exclude_from* _REGEX_
+
+# DESCRIPTION
+
+RPM uses a pluggable interface coupled with a file classifier to add dependency
+information to packages at build time. This system relies on specialized, domain
+specific executables, called _generators_, that do the heavy-lifting of figuring
+out the dependencies from the files present in the buildroot. It complements the
+manual tags, such as *Requires* or *Provides*, in *rpm-spec*(5) files.
+
+Generators are paired with logical labels, called _file attributes_, assigned to
+the packaged files by the file classifier. This ensures that every file is only
+processed by the applicable generator(s), as opposed to all generators, which
+simplifies their implementation and significantly improves the overall build
+performance.
```

Comment:
I'd drop the last part (", which ...") - man pages don't need to justify why something works the way it does or sell them to the users, they only need to tell you how it works.

Reply:
That's a good point. I felt the need to explain the *why* of file attributes since they technically aren't needed (they are just an optimization technique) and one might wonder what a file classifier has to do with all this. But then, you're also right - man pages aren't supposed to be text books, they're just that, manuals. Dry and to the point. So yep, will drop that.